### ISSUE TITLE
Reproduction problem

### ISSUE THREAD

**Issue by zhangxin-xd** [vimar-gu/MinimaxDiffusion#11]
Hi, thank you for sharing this excellent work!

I have some problems when reproducing the reported results. I directly ran the run.sh, the results of imagewoof under ipc=10 with resnet-ap is 36.2 (39.2 in  paper).

following is the arguments setting.

- finetuning
`Namespace(ckpt='pretrained_models/DiT-XL-2-256x256.pt', ckpt_every=12000, condense=True, data_path='/data/bobzhou/imagenet/train', epochs=8, finetune_ipc=-1, global_batch_size=8, global_seed=0, image_size=256, lambda_neg=0.008, lambda_pos=0.002, log_every=1500, memory_size=64, model='DiT-XL/2', nclass=10, num_classes=1000, num_workers=4, phase=0, results_dir='logs/run-0', spec='woof', tag='minimax', vae='ema')`
- generation
`Namespace(ckpt='pretrained_models/DiT-XL-2-256x256.pt', ckpt_every=12000, condense=True, data_path='/data/bobzhou/imagenet/train', epochs=8, finetune_ipc=-1, global_batch_size=8, global_seed=0, image_size=256, lambda_neg=0.008, lambda_pos=0.002, log_every=1500, memory_size=64, model='DiT-XL/2', nclass=10, num_classes=1000, num_workers=4, phase=0, results_dir='logs/run-0', spec='woof', tag='minimax', vae='ema')`
- validation
`Namespace(aug_type='color_crop_cutout', augment=True, batch_real=64, batch_size=64, batch_syn_max=128, beta=1.0, bias=False, bottleneck=True, data_dir='/data_large/readonly', dataset='imagenet', datatag='imagenet10', decode_type='single', depth=10, dsa=False, dsa_strategy='color_crop_flip_scale_rotate', dseed=0, early=0, epoch_print_freq=20, epochs=2000, f_idx='4', factor=1, fc=False, fix_iter=-1, imagenet_dir=['results/dit-distillation/imagenet-10-1000-minimax', '/data/bobzhou/imagenet/'], init='random', inner_loop=100, ipc=10, load_memory=True, lr=0.01, lr_img=0.005, match='grad', metric='l1', mix_p=1.0, mixup='cut', mixup_net='cut', modeltag='resnet10apin', mom_img=0.5, momentum=0.9, n_data=500, name='', nch=3, nclass=10, nclass_sub=-1, net_epoch=1, net_type='resnet_ap', niter=500, norm_type='instance', phase=-1, pretrained=False, print_freq=10, pt_from=-1, pt_num=1, repeat=1, reproduce=False, rrc=True, same_compute=False, save_ckpt=False, save_dir='./results/imagenet10/resnet10apin_test_rand10', seed=0, size=224, slct_type='random', spec='woof', tag='_test_rand10', test=False, time=False, verbose=False, weight_decay=0.0005, width=1.0, workers=8)`

Is there any mistakes? 

Looking forward to your response!

**Comment by vimar-gu** [comment]
Thanks for your interest in our work. Can you check if the result is better than your baseline implementation? The performance of small IPCs will be more unstable compared with larger ones. If it is better than the baseline, I will consider it the influence of randomness.

**Comment by zhangxin-xd** [comment]
Thank you for the prompt response! 

I am not sure what is the baseline  implementation. Do you mean higher ipc setting, like ipc=100?

Another point I’m curious about is: since the DiT model is already pretrained on ImageNet, why is it being fine-tuned on a subset? Is this primarily for improved performance? Additionally, I noticed in another discussion that the ImageNet results were achieved using an ensemble of 5 individual models.

**Comment by vimar-gu** [comment]
The baseline implementation is the images directly generated by DiT. 

Fine-tuning the DiT model on a subset was based on two considerations.
- In previous works, ImageWoof and other 10-class subsets are often used as the benchmark. So I followed this setting.
- The computational resources couldn't support me to further enlarge the training scale, especially regarding the mini-batch size. 

And the second reason leads to the hyper-parameter setting suitable for small mini-batch size fine-tuning, which is unfortunately not suitable for larger subsets. Therefore, I trained multiple models on different subsets. The ImageNet results are the combination of individually generated images from different models. I won't consider it an ensemble.

**Comment by zhangxin-xd** [comment]
Got it! Thank you for the explaination.  I will try. 😊

**Comment by zhangxin-xd** [comment]
Hi, thank you for your patience. I’ve run the baseline, and here are the results:

ImageWoof:
	•	IPC 10 with Minimax Fine-tuning: 36.2
	•	IPC 10 without Minimax Fine-tuning: 33.8

While Minimax fine-tuning has improved the baseline performance by 2.4 points, the results still show a noticeable gap compared to the reported value of 39.2 in the paper. The difference of 3.0 points exceeds the error bound of 1.3, and a similar situation persists in the IPC 100 setup.

I’d appreciate any insights or suggestions on aligning these results more closely with the reported ones. Thank you!

**Comment by vimar-gu** [comment]
Hi, in my experiments, the baseline yields a 34.7 accuracy, which is also higher than 33.8. I'm not sure about your experiment environments, but I can reproduce the results. How about changing the machine and random seeds?
